File: src/main/java/de/noname/pizza4c/datamodel/lieferando/OptionGroup.java
package de.noname.pizza4c.datamodel.lieferando;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class OptionGroup {
    @JsonProperty
    private String name;

    @JsonProperty
    private boolean isTypeMulti;

    @JsonProperty
    private boolean isRequired;

    @JsonProperty
    private int minChoices;

    @JsonProperty
    private int maxChoices;

    @JsonProperty
    private List<String> optionIds;

}

File: frontend/pizza4c-frontend/src/datamodel/optionGroup.ts
export interface OptionGroup {
    name: string;
    isTypeMulti: boolean;
    isRequired: boolean;
    minChoices: Number;
    maxChoices: Number;
    optionIds: string[];
}

export function isConfigurableOptionGroup(optionGroup: OptionGroup): boolean {
    return optionGroup.optionIds.length > 0;
}